packages/components/modal/src/ModalHeader/ModalHeader.tsx

```diff
@@ -52,6 +52,7 @@ export const ModalHeader = ({
         <Flex alignItems="center" className={styles.buttonContainer}>
           <Button
             variant="transparent"
+            className={styles.closeButton}
```

Comment:
I believe the correct appearance that @damann expects can be achieved by using `<IconButton size="small" icon={<CloseIcon size="small"} />`

just tried it quickly in the playground 

<img width="287" alt="image" src="https://github.com/contentful/forma-36/assets/1789174/841742a9-b4a3-47d7-a741-487890908c1d">

Reply:
@Lelith, thanks for the clarification, that makes sense.  I have replaced the previous `<Button>` with a `<IconButton>`, and confirmed that the button is now 16px from the edge of the modal (when the button is hovered).  I have reverted any style changes as well.  Lastly, I have a changeset file.

Please let me know if any other changes need to be made, thanks!

cc: @damann 

<img width="679" alt="Screenshot 2024-02-15 at 12 48 18 PM" src="https://github.com/contentful/forma-36/assets/158083968/36d6c8ba-eb42-4be0-a81d-737c23a7a635">

#### Hovered
<img width="669" alt="Screenshot 2024-02-15 at 12 52 13 PM" src="https://github.com/contentful/forma-36/assets/158083968/ab8043d0-8366-468a-81a5-1a22b541281a">